libs/SalesforceSDKCore/SalesforceSDKCore/Classes/Security/SFAuthenticationManager.m

```diff
@@ -429,11 +429,13 @@ - (void)logoutUser:(SFUserAccount *)user
         return;
     }
 
-    if (user.isUserLoggingOut) {
-        [self log:SFLogLevelInfo msg:@"logoutUser: user is already in the process of logout."];
-        return;
+    @synchronized (user) {
```

Comment:
I tend to be a bit more heavy handed and ensure all writes to the ivar are protected but you could be correct that this check is all that must be protected. Without knowing what the actual bizarre states the app could be in it is hard to say. For instance, could there be an authentication occurring at same time as a logout of the same user?